### PR TITLE
apply perf improvements from RTS

### PR DESCRIPTION
**Pull request by JoviDeCroock**
This puts us on par with what we had in the newer RTS versions

### CODE REVIEW

File: server/test/render.test.js [hunk 1]
```diff
@@ -860,7 +868,8 @@ describe('render', () => {
 	});
 
 	describe('state locking', () => {
-		it('should set _dirty to true', () => {
+		// TODO: we use flags now
+		it.skip('should set _dirty to true', () => {
```
Comment (JoviDeCroock) [inline server/test/render.test.js:872]: Skipped for now due to usage of flags

File: server/test/render.test.js [hunk 2]
```diff
@@ -1079,7 +1088,8 @@ describe('render', () => {
 		});
 	});
 
-	it('should invoke option hooks', () => {
+	// We call these with internals now
+	it.skip('should invoke option hooks', () => {
```
Comment (JoviDeCroock) [inline server/test/render.test.js:1092]: Skipped for now as we have no proper way to assert internals

File: server/test/render.test.js (hunk 1 above)
Comment (andrewiggins) [reply server/test/render.test.js:872]: Could we open an issue to track updating this or removing it if we determine it no longer makes sense?

**Review by andrewiggins — APPROVED** [review]
Woot!

File: server/test/render.test.js (hunk 1 above)
Comment (JoviDeCroock) [reply server/test/render.test.js:872]: Will do

**Review by marvinhagemeister — APPROVED** [review]

File: server/test/render.test.js (hunk 1 above)
Comment (JoviDeCroock) [reply server/test/render.test.js:872]: Opened #3859

**Pull request merged**